Sources/PackageManagerDocs/Documentation.docc/Dependencies/PackageTraits.md

```diff
@@ -0,0 +1,138 @@
+# Provide configurable packages using traits.
+
+Define one or more traits to offer default and configurable features for a package.
+
+Prior to Swift packages prior to Swift 6.1 offered a non-configurable API surface for each version.
+With Swift 6.1, packages may offer traits, which express a configurable API surface for a package.
+
+Use traits to enable additional API beyond the core API of the package.
+For example, a trait may enable an experimental API, optional extended functionality that requires additional dependencies, or functionality that isn't critical that a developer may want to disable in specific circumstances.
+
+> Note: Traits should always *enable* additional code, never "remove" or disable API when a trait is enabled.
+
+Within the package, traits express conditional compilation, and may be used to declare additional dependencies that are enabled when that trait is active.
+
+Traits are identified by their names, which are name-spaced within the package that hosts them.
```

Comment:
IIRC the latest is that traits use be valid Swift identifiers. We can link to the Swift book's grammar section on this.